python/python/bystro/proteomics/fragpipe_utils.py

```diff
@@ -0,0 +1,21 @@
+"""Provide various utilities for Fragpipe analysis."""
+
+import pandas as pd
+
+ANNOTATION_COLS = ["plex", "channel", "sample", "sample_name", "condition", "replicate"]
+
+
+def check_df_cols(df: pd.DataFrame, expected_cols: list[str]) -> None:
+    """Check that df cols contain expected cols as a prefix."""
+    actual_cols = df.columns
+    if not all(x == y for x, y in zip(expected_cols, actual_cols, strict=False)):
```

Comment:
Will expected_cols contain the minimal set of column headers that we're expected for a given proteomics 'annotation' file? I may be missing context but how/where will we handle columns specific to an annotation type file that are experiment-specific (will this type of information be separated into subject-level information for instance)?

Reply:
That's a good question and the short answer is I don't know yet, but I think it falls out of scope for this method, which is doing something more modest.  I've renamed the method though to try to make it clearer what it's supposed to be doing, though.

Reply:
We definitely don't have the subject level info story completely worked out yet: for now everything is more or less its own independent sample